src/language/semantics/xsltSnippets.ts

```diff
@@ -0,0 +1,178 @@
+/*---------------------------------------------------------------------------------------------
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface Snippet {
+  name: string
+  body: string
+  description: string
+}
+
+export class XSLTSnippets {
+  static xsltRootTags: Snippet[] = [
```

Comment:
@rthomas320 Please comment on this

@stevedlawrence I did add these files to the LICENSE and bin.LICENSE as well

Reply:
or @michael-hoke can you explain what this is needed for?

Reply:
short answer - maybe not?

longer answer - The XSLT-related files are pretty integrated into the code that tokenizes what we identify as XPath, so we may or may not be able to remove the XSLT-related files. The main deterrent for this, from what I remember, was the XsltTokenDiagnostics invocation needed right after we identify the sections of the DFDL schema that should be treated as XPath (for colorization/highlighting).

Also, random side-note, but based on the name and position of the XsltTokenDiagnostics call, I think it might display error messages in vscode if there are any syntax errors in the XPath (unknown if this is actually the case or not). If this is the case, it might be that the files that are XSLT-related are dual-purpose between XSLT and XPath.

I'd be happy to look into either removing the XSLT files or at least pulling out the strictly-XSLT pieces and renaming them XPath/XP, but I don't think this will be a quick fix, so can we consider creating an issue and looking into it separately from this PR?

Reply:
The bin.LICENSE has the github repo URL that it came from. Removing the XPath may cause more issues that would take a while to remove these and either replace them using a node package or just having them gone completely. I think it maybe be best to let them in as a starting point then removing them in another PR in favor of a node package

Reply:
@stevedlawrence @rt320 @michael-hoke I was able to delete the code copied from the two VS Code repo and just add these extension as dependencies so whenever our extension is installed it will require both of these extensions. Everything seemed to work perfectly how it did with all the code copied in

Reply:
Unless I'm missing something, we no longer have semantic colorization now? So that XPath expressions in DFDL attributes/elements are just solid string colored? If that's the desired behavior, remove DeltaXML as a dependency because we're not using it for anything else, as far as I can remember. We weren't using the whole DeltaXML plugin codebase anyway - the code that was taken was a minimum amount to grab to get semantic highlighting to work (might be able to reduce that further by taking out XSLT as discussed above).

Doing this for the tag highlighting extension is fine I think.

Reply:
So how do I test if the semantic colorization still working? I could never tell which one was which was from the tag highlighting and the DeltaXML stuff. Everything seemed to be the exact same so I thought the semantic colorization was working

Reply:
I have been using the official PNG schema for my testing. https://github.com/DFDLSchemas/PNG/blob/master/src/main/resources/com/mitre/png/xsd/png.dfdl.xsd

Starting on Line 53, you can see a section of XPath. Without semantic highlighting, this whole section will be highlighted as if it were just another string. Unless the semantic highlighting is just running on the whole file and ignoring the syntactic highlighting?

For the tag highlighting, the yellow underline should show up, and I added code for the status bar that showed the current element and all parents - not sure if that is done automatically or not.

Reply:
@michael-hoke So for example using the file `dfdl-ethernet/src/main/resources/com/owlcyberdefense/dfdl/xsd/ethernetIP.dfdl.xsd` it has

```xml
  <xs:complexType name="Ethernet">
    <xs:sequence>
      <xs:element name="MACDest" type="b:hexByte" dfdl:length="6" />
      <xs:element name="MACSrc" type="b:hexByte" dfdl:length="6" />
      <xs:element name="Ethertype" type="b:bit" dfdl:length="16"
                  dfdl:outputValueCalc="{
          if (fn:exists(../NetworkLayer/IPv4)) then 2048
          else if (fn:exists(../NetworkLayer/IPv6 )) then 34525
          else fn:error('ethernet', 'fn:error called.', .) }"/>
      <xs:element name="NetworkLayer" type="tns:NetworkLayer" />
    </xs:sequence>
  </xs:complexType>
```
So the semantic colorization should show these if statements as a different color right? Not just color it the same way it does in this comment correct? If so then it is still working just installing that extension as mine shows the if - else if - else portion as pure white

Reply:
Thanks for sending me the vsix files. Unfortunately, neither of them are working the way I would expect them to... So, this stopped working at some point...

The way I see it, we have a few options here:
- Keep the code as we have it currently, which doesn't highlight embedded XPath, remove the dependency on DeltaXML. Mark that this is the expected behavior.

- Keep the code as we have it currently, open an issue on getting this functionality back in separately from the PR (which would involve looking into taking out the XSLT files). This would mean including a subset of the DeltaXML plugin, which would require us to figure out the licensing information for those. A first step for this would be to add the dependency to deltaXML back in but figure out how to give the plugin the XPath segments so that we wouldn't have to include that code in our extension.

- Revert the commit, remove only the tag-highlighting, and get this working as a part of this PR (with or without potential XSLT pruning).

Reply:
Yeah I am thinking if I should remove that deltaXML dependency, I also needed to update .vscode ignore to add the omega_edit/interface.html file, and then we make a ticket for adding the XPath embedded highlighting? I don't think we want to add the files and have an issue to look into removing them think it would just be better to make a separate PR that will either add those files in or use the extension as a dependency but still have it work the same way. @stevedlawrence What do you think?